### ISSUE TITLE
Warnings when building under Windows

### ISSUE THREAD

**Issue by lokitoth** [VowpalWabbit/vowpal_wabbit#1820]
#### Describe the bug
* Python bindings builds with warnings on Windows
* vw_clr builds with warnings on Windows (CodeAnalysis issue)
* GitLink causes warnings with build-time generated files on Windows
* warm_cb.cc builds with warning on Windows
* simulator builds with warnings on Windows

#### To Reproduce
1. Clone the repo
2. In a VS Developer Command Prompt run:
    a. .scripts\restore.cmd
    b. .scripts\build.cmd

#### Expected behavior
No build warnings in Windows build

#### Observed Behavior

From https://ci.appveyor.com/project/JohnLangford/vowpal-wabbit/builds/23526080, lines 992 - 1020

```
992
993
994
       "C:\vowpal_wabbit\vowpalwabbit\vw.sln" (default target) (1) ->
995
       "C:\vowpal_wabbit\python\windows27\python27.vcxproj" (default target) (15) ->
996
       (ClCompile target) -> 
997
         C:\vowpal_wabbit\vowpalwabbit\packages\boost.1.63.0.0\lib\native\include\boost/python/detail/caller.hpp(55): warning C4244: 'return': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data [C:\vowpal_wabbit\python\windows27\python27.vcxproj]
998
999
1000
       "C:\vowpal_wabbit\vowpalwabbit\vw.sln" (default target) (1) ->
1001
       "C:\vowpal_wabbit\cs\unittest\cs_unittest.csproj" (default target) (6) ->
1002
       "C:\vowpal_wabbit\cs\cs_console\cs_console.csproj" (default target) (8) ->
1003
       "C:\vowpal_wabbit\cs\cli\vw_clr.vcxproj" (default target) (9) ->
1004
       (RunCodeAnalysis target) -> 
1005
         MSBUILD : warning : CA0064 : No analysis was performed because the specified rule set could not be loaded or did not contain any managed code analysis rules. [C:\vowpal_wabbit\cs\cli\vw_clr.vcxproj]
1006
1007
1008
       "C:\vowpal_wabbit\vowpalwabbit\vw.sln" (default target) (1) ->
1009
       "C:\vowpal_wabbit\cs\unittest\cs_unittest.csproj" (default target) (6) ->
1010
       "C:\vowpal_wabbit\cs\cs\cs.csproj" (default target) (7) ->
1011
       (LinkPdbToGitRemote target) -> 
1012
         C:\vowpal_wabbit\vowpalwabbit\packages\GitLink.3.1.0\build\GitLink.targets(9,5): warning : No source files found that are tracked in a git repo. [C:\vowpal_wabbit\cs\cs\cs.csproj]
1013
1014
1015
       "C:\vowpal_wabbit\vowpalwabbit\vw.sln" (default target) (1) ->
1016
       "C:\vowpal_wabbit\cs\examples\simulator\simulator.csproj" (default target) (17) ->
1017
       (ResolveAssemblyReferences target) -> 
1018
         C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1820,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\vowpal_wabbit\vowpalwabbit\out\target\Release\VowpalWabbit.Common.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [C:\vowpal_wabbit\cs\examples\simulator\simulator.csproj]
1019
         C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1820,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\vowpal_wabbit\vowpalwabbit\out\target\Release\x64\VowpalWabbit.Core.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [C:\vowpal_wabbit\cs\examples\simulator\simulator.csproj]
1020
         C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1820,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\vowpal_wabbit\vowpalwabbit\out\target\Release\VowpalWabbit.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [C:\vowpal_wabbit\cs\examples\simulator\simulator.csproj]
```
#### Environment
What version of VW did you use? VowpalWabbit/vowpal_wabbit:master
What OS or language did you use? Windows / VS 2017 (but is reproducible in AppVeyor)

#### Additional context
Is there some way to add warning/error highlighting to AppVeyor builds?

**Comment by lokitoth** [comment]
Partially fixed by #2279

**Comment by jackgerrits** [comment]
Closing as @olgavrou fixed most/all of the Windows warnings recently.
